src/setLocalStore.ts

```diff
@@ -1,44 +1,21 @@
-/**
- * ************************************
- *
- * @module next-persist
- * @author most-js
- * @description module that contains function to write state to localStorage
- *
- * ************************************
- */
+import { NextPersistConfig, LooseObject } from './types';
 
-interface LooseObject {
-  // eslint-disable-next-line  @typescript-eslint/no-explicit-any
-  [key: string]: any;
-}
-
-interface AllowListObject {
-  [key: string]: string[];
-}
-
-export function setLocalStore(
-  config: AllowListObject,
-  state: LooseObject
-): void | { [key: string]: string } {
-  const key = Object.keys(config)[0];
-  const allowList = Object.values(config)[0];
-
-  // if application is running client-side, localStorage is accessible
+const setLocalStore = (config: NextPersistConfig, state: LooseObject): void => {
```

Comment:
@grishaLR looks good. would it be overkill to have allow list account for deeply nested props?

Reply:
this sounds like it would be a good enhancement and should probably implement. I'm going to see how I could add that in a future PR. I'd like to keep this one low risk